spec/device_detector/model_extractor_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'spec_helper'
+
+describe DeviceDetector::ModelExtractor do
+
+  subject(:extractor) { DeviceDetector::ModelExtractor.new(user_agent, regex_meta) }
+
+  describe '#call' do
+
+    context 'when matching against dynamic model' do
+
+      let(:regex_meta) do
+        {
+          'regex'  => '(?:Apple-)?iPhone ?(3GS?|4S?|5[CS]?|6(:? Plus)?)?',
+          'model'  => 'iPhone $1',
+          'device' => 'smartphone'
+        }
+      end
+
+      context 'when no dynamic match is found' do
+        let(:user_agent) { 'Mozilla/5.0 (iPhone; CPU iPhone OS 8_1_3 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) Version/8.0 Mobile/12B466 Safari/600.1.4' }
+        let(:device_name) { 'iPhone' }
+
+        it 'should return the textual portion without trailing whitespace' do
+          expect(extractor.call).to eq(device_name)
```

Comment:
How about putting explicitly `iPhone` without the `let`? It would make it clearer as what to expect, plus it is only used in this line, actually.